pkg/awsutils/awsutils.go

```diff
@@ -845,33 +860,123 @@ func (cache *EC2InstanceMetadataCache) createENI(useCustomCfg bool, sg []*string
 			SecondaryPrivateIpAddressCount: aws.Int64(int64(needIPs)),
 		}
 	}
-	if useCustomCfg {
-		log.Info("Using a custom network config for the new ENI")
-		if len(sg) != 0 {
-			input.Groups = sg
+
+	var err error
+	var networkInterfaceID string
+	if cache.useCustomNetworking {
+		input = createENIUsingCustomCfg(sg, eniCfgSubnet, input)
+		log.Infof("Creating ENI with security groups: %v in subnet: %s", aws.StringValueSlice(input.Groups), aws.StringValue(input.SubnetId))
+
+		networkInterfaceID, err = cache.tryCreateNetworkInterface(input)
+		if err == nil {
+			return networkInterfaceID, nil
+		}
+	} else {
+		if cache.useSubnetDiscovery {
+			subnetResult, vpcErr := cache.getVpcSubnets()
+			if vpcErr != nil {
```

Comment:
Maybe as a fast follow up lets restructure the code..

1. Get subnet -> Either primary ENI, ENIConfig or pick a subnet
2. Get SG -> Either primary ENI SG or ENIConfig SG
3. Modify the "input" with the new subnet ID and SG ID
4. Call tryCreateNetworkInterface